### DISCUSSION TITLE
Endpoints : Default Parameter Values Not Working

### DISCUSSION THREAD

**Discussion by tricaricom** [serverpod/serverpod#2768] [Q&A]
I have an endpoint like this:

`Future<void> addMapData(Session session, MapData mapData, { bool sendAlerts = true }) async { ... }`

But on the Flutter side, it gives me the error that `sendAlerts` is a required parameter.  It looks like `required` was added:

`{ required bool sendAlerts }`

Is this by design, or am I doing something wrong?

**Comment by hampuslavin** [comment]
Hi @tricaricom! 👋 
Thank you for reporting this! There are two related issues at play here:
1. The `sendAlerts` variable is interpreted as `required` which is incorrect
2. The default value is not transferred to the client

I have created issues for them here:
- https://github.com/serverpod/serverpod/issues/2788
- https://github.com/serverpod/serverpod/issues/2787
